# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1272124198 14400
#      Sat Apr 24 11:49:58 2010 -0400
# Node ID 5f46cc552ce5fe4e449292c72679a3fa823dbd5b
# Parent  ea79ebe99051aadc74cff5d094ab02f405ff0d50
__go_draw_axes__.m: Consistent placement of axes for ps vs other gnuplot terminals.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-04-24  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Consistent placement of axes for ps vs
+	other gnuplot terminals.
+
 2010-04-23  Ben Abbott <bpabbott@mac.com>
 
 	* plot/colorbar.m: Fix colorbar() bugs for manual plotboxaspectratio.
 	* plot/axis.m: For 'axis normal' restore plotboxaspectratio.
 
 2010-04-23  John W. Eaton  <jwe@octave.org>
 
 	* gethelp.cc (extract_help_text): Use C++ static_cast instead of
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -45,18 +45,18 @@ function __go_draw_axes__ (h, plot_strea
       ymirror = true;
     endif
 
     nd = __calc_dimensions__ (axis_obj);
     if (strcmpi (axis_obj.plotboxaspectratiomode, "manual"))
       pos = __actual_axis_position__ (axis_obj);
     else
       pos = axis_obj.position;
-      pos = pos - implicit_margin([1, 2, 1, 2]).*[1, 1, -0.5, -0.5];
     endif
+    pos(1:2) = pos(1:2) - implicit_margin .* [0.75, 0.5];
     if (__gnuplot_has_feature__ ("screen_coordinates_for_{lrtb}margin"))
       if (nd == 2)
         x = [1, 1];
       else
         ## 3D plots need to be sized down to fit in the window.
         x = 1.0 ./ sqrt([2, 2.5]);
       endif
       fprintf (plot_stream, "set tmargin screen %.15g;\n", pos(2)+pos(4)/2+x(2)*pos(4)/2);
